src/gadgets/mpt_update.rs

```diff
@@ -1525,12 +1494,6 @@ fn configure_poseidon_code_hash<F: FieldExt>(
     cb: &mut ConstraintBuilder<F>,
     config: &MptUpdateConfig,
 ) {
-    cb.assert(
```

Comment:
ExtensionNew(Old) are already disallowed for PoseidonCodeHashExists in configure_extension_new(old).